presto-main-base/src/main/java/com/facebook/presto/metadata/HandleResolver.java

```diff
@@ -98,7 +145,13 @@ public String getId(ColumnHandle columnHandle)
 
     public String getId(ConnectorSplit split)
     {
-        return getId(split, MaterializedHandleResolver::getSplitClass);
+        try {
+            return getId(split, MaterializedHandleResolver::getSplitClass);
+        }
+        catch (Exception e) {
+            // Fallback if needed
+            return getFunctionId(split, tableFunctionSplitResolvers);
```

Comment:
Can you explain why we need this Exception-based fallback?  Is there any way to refactor to avoid the need to do this?  It's far preferable to just let Exceptions bubble up.